### PR TITLE
Dev subcategories card21 product card37

### PR DESCRIPTION
**Pull request by vita792**

### CODE REVIEW

File: src/main/java/.env
Comment (halyna-k) [inline src/main/java/.env:1]: You cannot upload .env file to GitHub. It needs to be added to .gitignore

File: src/main/java/com/megastore/controller/APIController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return new Message("All good. You can see this because you are Authenticated.");
+    }
+
+    @GetMapping(value = "/private-scoped")
+    public Message privateScopedEndpoint() {
+
+        return new Message("All good. You can see this because you are Authenticated with a Token granted the 'read:users' scope");
+    }
+
+    @GetMapping("/home/hotProducts")
```
Comment (dimalian) [inline src/main/java/com/megastore/controller/APIController.java]: I would rename this endpoint to "/product/hot". If you retrieve product data you should have the same route name "product". You can add something like "hot" to the end if you try to get some special type of product.
Other examples:
retrieve one product by id: GET "/product/:id"
retrieve a list of products: GET "/product/list"
delete a product: DELETE "/product"
add a new product: POST "/product"
update a product: PATCH "/product"

File: src/main/java/com/megastore/controller/ApplicationController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@RestController
+@RequestMapping("/api/v1")
+public class ApplicationController {
+
+    @GetMapping("/home")
+    public String home() {
+        return new String("Hello MegaStore!");
+    }
+
+    @GetMapping("/home/hotProducts")
```
Comment (dimalian) [inline src/main/java/com/megastore/controller/ApplicationController.java]: same here

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -8,4 +8,13 @@ spring.datasource.driver-clas-name=org.postgresql.Driver
...
+spring.datasource.validationQuery=SELECT 1
+
+#Security
+#Google login
+spring.security.oauth2.client.registration.google.client-id=520695556063-hi6gtn4715qn7ob251045a328cnlmbaj.apps.googleusercontent.com
+spring.security.oauth2.client.registration.google.client-secret=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
+
+#Facebook login
+spring.security.oauth2.client.registration.facebook.client-id=917487869325725
+spring.security.oauth2.client.registration.facebook.client-secret=156a7024a0c18d0afcfd14ef318f623e
```
Comment (dimalian) [inline src/main/resources/application.properties]: Values for this file you need to retrieve from the .env file

File: src/main/resources/database/data.sql
Comment (dimalian) [inline src/main/resources/database/data.sql:1]: don't push empty files

File: src/main/resources/database/schema.sql
Comment (dimalian) [inline src/main/resources/database/schema.sql:1]: don't push empty files

**Review by halyna-k — APPROVED** [review]

**Review by dimalian — APPROVED** [review]

**Pull request merged**